### PR TITLE
Rewrite cursor example to include speedup possibilities

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- Cleaned up the two existing implementations a bit.
- Hide the cross hair when the cursor leaves the Axes (IMHO more reasonable than letting it stay visible at some random position near the border).
- Sped up the SnappingCursor by only redrawing if the indicated data point changes.
- Added an implementation using blitting.

### CODE REVIEW

File: examples/misc/cursor_demo_sgskip.py [hunk 1]
```diff
@@ -22,70 +28,154 @@
...
+# Faster redrawing using blitting
+# """""""""""""""""""""""""""""""
+# This technique stores the rendered plot as a background image. Only the
+# changed artists (cross hair lines and text) are rendered anew. They are
+# combined with the background using blitting.
+#
+# This technique is significantly faster. It requires a bit more setup because
+# the background has to be stored without the cross hair lines (see
+# ``create_background()``). Additionally, a new background has to be created
+# whenever the figure changes. The example implements this for resizing. If
```
Comment (QuLogic) [inline examples/misc/cursor_demo_sgskip.py]: This also doesn't handle pan & zoom, which are enabled by default.

File: examples/misc/cursor_demo_sgskip.py [hunk 2]
```diff
@@ -22,70 +28,154 @@
...
+ax.set_title('Blitted cursor')
+ax.plot(x, y, 'o')
+blitted_cursor = BlittedCursor(ax)
+fig.canvas.mpl_connect('motion_notify_event', blitted_cursor.mouse_move)
+
+
+##############################################################################
+# Snapping to data points
+# """""""""""""""""""""""
+# The following cursor snaps it's position to the data points of a `.Line2D`
```
Comment (QuLogic) [inline examples/misc/cursor_demo_sgskip.py]:
```suggestion
# The following cursor snaps its position to the data points of a `.Line2D`
```

File: examples/misc/cursor_demo_sgskip.py (hunk 1 above)
Comment (timhoffm) [reply examples/misc/cursor_demo_sgskip.py]: I know. Thought I could get away with that.

I've now changed to recreating the background on every draw. This is a bit more fiddly but covers all possible changes.

File: examples/misc/cursor_demo_sgskip.py [hunk 3]
```diff
@@ -22,70 +28,192 @@
...
-        self.txt.set_text('x=%1.2f, y=%1.2f' % (x, y))
-        self.ax.figure.canvas.draw()
-
-
-class SnaptoCursor:
+fig, ax = plt.subplots()
+ax.set_title('Simple cursor')
+ax.plot(x, y, 'o')
+cursor = Cursor(ax)
+fig.canvas.mpl_connect('on_mouse_move', cursor.mouse_move)
```
Comment (anntzer) [inline examples/misc/cursor_demo_sgskip.py]: `cursor.on_mouse_move` (below two more times as well)

File: examples/misc/cursor_demo_sgskip.py [hunk 4]
```diff
@@ -22,70 +28,192 @@
...
+# moves far enough so that another data point must be selected. This reduces
+# the lag due to many redraws. Of course, blitting could still be added on top
+# for additional speedup.
+
+class SnappingCursor:
+    """
+    A cross hair cursor that snaps to the data point of a line, which is
+    closest to the *x* position of the cursor.
 
+    For simplicity, this assumes that *x* values of he data is sorted.
```
Comment (anntzer) [inline examples/misc/cursor_demo_sgskip.py]: "the data" (typo)

**Pull request merged**